### PR TITLE
Adding Thor to coconut, so it can be an executable gem

### PR DESCRIPTION
**Pull request by royGap**

### CODE REVIEW

File: lib/coconut.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 require 'yaml'
 
-module Coconut
-end
+# module Coconut
```
Comment (royGap) [inline lib/coconut.rb]: remove this comments

File: lib/coconut/version.rb [hunk 2]
```diff
@@ -1,3 +1 @@
-module Coconut
-  VERSION = "0.1.0"
-end
+VERSION = "0.1.0"
```
Comment (royGap) [inline lib/coconut/version.rb]: bump the version to 0.1.1

File: lib/generators/coconut/templates/coconut.tt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+local:
+  customer_path: customers
```
Comment (royGap) [inline lib/generators/coconut/templates/coconut.tt]: remove this element is no longer needed

File: README.md [hunk 4]
```diff
@@ -8,43 +8,33 @@ Coconut fetch the configuration files from development servers and stored them o
...
-And then execute:
-
-    $ bundle
-
-Or install it yourself as:
-
-    $ gem install coconut
 
+The following command will init a coconut instance on your proyect.
+```Init a coconut instanca
```
Comment (jpaparra) [inline README.md]: instance*

File: lib/coconut/cache_service.rb [hunk 5]
```diff
@@ -1,27 +1,25 @@
-module Coconut
-  class CacheService
+class CacheService
 
-    TMP_PATH = "#{Rails.root}/tmp"
+  TMP_PATH = "./tmp"
```
Comment (jpaparra) [inline lib/coconut/cache_service.rb]: single quotes

**Review by jpaparra — CHANGES_REQUESTED** [review]

File: lib/coconut/commands/fetch.rb [hunk 6]
```diff
@@ -1,5 +1,6 @@
-module Coconut
-  class FetchService
+module Commands
```
Comment (andresgap) [inline lib/coconut/commands/fetch.rb]: Does commands represent the class? Seems to generic

File: lib/generators/init_file.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+require 'thor'
+require 'thor/group'
+
+module Generators
+  class InitFile < Thor::Group
+    include Thor::Actions
+
+    desc 'Generate configuration to init a coconut instance on a SQ proyect'
```
Comment (andresgap) [inline lib/generators/init_file.rb]: Remove any referece to SQ or D360

File: bin/coconut [hunk 8]
```diff
@@ -1,6 +1,27 @@
...
-puts "This is your coconut CLI."
-puts __FILE__
+class Coconut < Thor
+
+  desc "install", "install the coconut on the local enviroment"
+  def install
+    Commands::Install.new.start
+  end
+
+  desc "init", "init a SQ project with a coconut instance"
```
Comment (andresgap) [inline bin/coconut]: Same here

File: lib/coconut/cli.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
+module Coconut
+
+  class CLI < Thor
+
+    desc "install", "install the coconut on the local enviroment"
```
Comment (jpaparra) [inline lib/coconut/cli.rb]: Use single quotes

File: lib/generators/coconut/templates/coconut.tt [hunk 10]
```diff
@@ -11,7 +11,6 @@ local:
       swap: true
     star_schema.xml:
       swap: false
-
```
Comment (andresgap) [inline lib/generators/coconut/templates/coconut.tt]: Please remove anything related to specific server configurations, like these file names or paths

**Pull request merged**